auto-self-nominate/index.ts

```diff
@@ -0,0 +1,166 @@
+    }
+    return c;
+}
+
+const networkId = getConfig("NETWORK_ID");
+const rpcUrl = getConfig("RPC_URL");
+
+const sdk = (() => {
+    console.log(`sdk ${networkId} ${rpcUrl}`);
+    return new SDK({
+        server: networkId,
+        networkId: rpcUrl,
+    });
+})();
+
+function decodePlatformAddressFromPubkey(buffer: Buffer): PlatformAddress {
+    const pubkey = buffer.toString("hex");
+    return PlatformAddress.fromPublic(pubkey, {
+        networkId: sdk.networkId,
+    });
+}
+
+function decodeNumber(buffer: Buffer): number {
+    return parseInt(buffer.toString("hex"), 16);
+}
+
+async function getCurrentTermId(bestBlockNumber: number): Promise<number | null> {
+    return new Promise((resolve, reject) => {
+        sdk.rpc
+            .sendRpcRequest("chain_getTermMetadata", [bestBlockNumber])
+            .then(result => {
+                if (result === null) {
+                    resolve(null);
+                }
+                const [prevTermLastBlockNumber, currentTermId] = result;
+                if (
+                    typeof prevTermLastBlockNumber === "number" &&
+                    typeof currentTermId === "number"
+                ) {
+                    resolve(currentTermId);
+                }
+                reject(
+                    Error(
+                        `Expected getTermMetadata to return [number, number] | null but it returned ${result}`,
+                    ),
+                );
+            })
+            .catch(reject);
+    });
+}
+
+async function sendSelfNominateTransaction(
+    accountInfo: { account: string; passphrase: string },
+    params: { deposit: U64; metadata: string },
+) {
+    const tx = createSelfNomination(params);
+    const { account, passphrase } = accountInfo;
+    await sdk.rpc.account.sendTransaction({
+        tx,
+        account,
+        passphrase,
+    });
+}
+
+function createSelfNomination(params: { deposit: U64; metadata: string }): Custom {
+    const { deposit, metadata } = params;
+    const handlerId = new U64(STAKE_ACTION_HANDLER_ID);
+    const actionTag = ACTION_TAG_SELF_NOMINATE;
+    const bytes = RLP.encode([actionTag, deposit, metadata]);
+    return new Custom(
+        {
+            handlerId,
+            bytes,
+        },
+        networkId,
+    );
+}
+
+async function getCandidatesInfo(bestBlockNumber: number): Promise<CandidatesInfo> {
+    const retval: CandidatesInfo = {};
+    const data = (await sdk.rpc.engine.getCustomActionData(2, ["Candidates"], bestBlockNumber))!;
+    const list: Buffer[][] = RLP.decode(Buffer.from(data, "hex"));
+    list.forEach(
+        ([encodedPubkey, encodedDeposit, encodedNominationEnd]) =>
+            (retval[decodePlatformAddressFromPubkey(encodedPubkey).value] = [
+                decodeNumber(encodedNominationEnd),
+                decodeNumber(encodedDeposit),
+            ]),
+    );
+    return retval;
+}
+
+async function supplementaryNoimationDeposit(
```

Comment:
This function does two things. One is checking whether it should send a nominate transaction, and the other is calculating the deposit. How about renaming the function or split the function?

Reply:
I split the function into two different functions.